roles/openshift_master/templates/master.yaml.v1.j2

```diff
@@ -18,8 +18,8 @@ assetConfig:
 {% if 'logging_public_url' in openshift.master %}
   loggingPublicURL: {{ openshift.master.logging_public_url }}
 {% endif %}
-{% if 'metrics_public_url' in openshift.master %}
-  metricsPublicURL: {{ openshift.master.metrics_public_url }}
+{% if openshift_hosted_metrics_deploy | bool %}
+  metricsPublicURL: https://hawkular-metrics.{{ openshift.master.default_subdomain }}/hawkular/metrics
```

Comment:
The `openshift_metrics` role is setting the URL like the above changes. Both configs should feed from the same variable while providing compatibility with `openshift_master_metrics_public_url` (if changed). I expect we'd want the variable to be `openshift_hosted_metrics_public_url` if we allowed overriding in either case and we could default to this value.


Within the `openshift_metrics` role:
```
HAWKULAR_METRICS_HOSTNAME=hawkular-metrics.{{ openshift.master.default_subdomain }}
```